plugins/catalog-backend/src/ingestion/types.ts

```diff
@@ -68,3 +69,75 @@ export type ReadLocationError = {
   location: LocationSpec;
   error: Error;
 };
+
+//
+// LocationAnalyzer
+//
+
+export type LocationAnalyzer = {
+  /**
+   * Generates an entity configuration for given git repository. It's used for
+   * importing new component to the backstage app.
+   *
+   * @param location Git repository to analyze and generate config for.
+   */
+  analyzeLocation(
+    location: AnalyzeLocationRequest,
+  ): Promise<AnalyzeLocationResponse>;
+};
+
+export type AnalyzeLocationRequest = {
+  location: LocationSpec;
+};
+
+export type AnalyzeLocationResponse = {
+  existingEntityFiles: AnalyzeLocationExistingEntity[];
+  generateEntities: AnalyzeLocationGenerateEntity[];
+};
+
+// If the folder pointed to already contained catalog info yaml files, they are
+// read and emitted like this so that the frontend can inform the user that it
+// located them and can make sure to register them as well if they weren't
+// already
+type AnalyzeLocationExistingEntity = {
+  location: LocationSpec;
+  isRegistered: boolean;
+  entity: Entity;
+};
+
+// This is some form of representation of what the analyzer could deduce.
+// We should probably have a chat about how this can best be conveyed to
+// the frontend. It'll probably contain a (possibly incomplete) entity, plus
+// enough info for the frontend to know what form data to show to the user
+// for overriding/completing the info.
+type AnalyzeLocationGenerateEntity = {
+  // Some form of partial representation of the entity
+  entity: RecursivePartial<Entity>;
+  // Lists the suggestions that the user may want to override
+  fields: AnalyzeLocationEntityField[];
+};
+
+// This is where I get really vague. Something like this perhaps? Or it could be
+// something like a json-schema that contains enough info for the frontend to
+// be able to present a form and explanations
+type AnalyzeLocationEntityField = {
+  // e.g. "spec.owner"? The frontend needs to know how to "inject" the field into the
+  // entity again if the user wants to change it
+  field: string;
+
+  // The outcome of the analysis for this particular field
+  state:
+    | 'analysisSuggestedValue'
+    | 'analysisSuggestedNoValue'
+    | 'needsUserInput';
+
+  // If the analysis did suggest a value, this is where it would be. Not sure if we want
+  // to limit this to strings or if we want it to be any JsonValue
+  value: string | null;
+
+  // A text to show to the user to inform about the choices made. Like, it could say
+  // "Found a CODEOWNERS file that covers this target, so we suggest leaving this
+  // field empty; which would currently make it owned by X" where X is taken from the
+  // codeowners file.
+  description: string;
```

Comment:
In the `CodeOwnersProcessor` there is a function which throws an error if the codeowners file is not present, so I assume this is deliberate that these warning are being logged. It's severity level is warning so it should be ok. It's definitely not related to this pull request (it started to occur now because I recently removed the `owner` field from generated configs and CodeOwnersProcessor isn't even invoked if owners field in the configuration is present)

Reply:
Yes, that makes sense. If the `owner` field is already present then the `CodeOwnersProcessor` will not attempt to fetch the owner via a `CODEOWNERS` file. This brings up an interesting question though. Some organizations (like ours) use the `CODEOWNERS` extensively likely and others do not. So if this plugin does not include an `owner` it will be an ideal experience for some folks but will actually not work end-to-end for others as the `owner` is a required field.

Reply:
The idea in the end is that the import will be smart enough to be able to say "i found codeowners so I'll recommend leaving this empty (but feel free to override me)", vs "i was not able to find a sensible user so you'll have to fill one in in a form in the frontend before posting submit "